src/System.Management.Automation/engine/ComInterop/ConvertibleArgBuilder.cs

```diff
@@ -1,36 +1,22 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT License.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
 
-#if !SILVERLIGHT
-
-#if !CLR2
+using System;
 using System.Linq.Expressions;
-#else
-using Microsoft.Scripting.Ast;
-#endif
-//using Microsoft.Scripting.Utils;
-using System.Management.Automation.Interpreter;
 
 namespace System.Management.Automation.ComInterop
 {
     internal class ConvertibleArgBuilder : ArgBuilder
     {
-        internal ConvertibleArgBuilder()
-        {
-        }
-
         internal override Expression Marshal(Expression parameter)
         {
             return Helpers.Convert(parameter, typeof(IConvertible));
         }
 
         internal override Expression MarshalToRef(Expression parameter)
         {
-            // we are not supporting convertible InOut
-            throw Assert.Unreachable;
+            //we are not supporting convertible InOut
+            throw new NotSupportedException();
```

Comment:
PSNotSupportedException?

Reply:
Same here. I prefer to keep the code as is, so the diffs between the code here and in dotnet/runtime is minimal.